org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/functions/InputFunction.java

```diff
@@ -43,7 +45,10 @@ public InputFunction(IResultFactory resultFactory, @Assisted IProject project,
             throw new MetaborgException("Cannot find the shell facet.");
         }
 
-        FileObject file = project.location().resolveFile("temp");
+        ResourceExtensionFacet extensionFacet = lang.facet(ResourceExtensionFacet.class);
```

Comment:
I remember using `lang.facet` for the ShellFacet could return null, so if I'm not mistaken this requires a null check.